hack/ebs-scale-test/scale-test

```
@@ -0,0 +1,108 @@
+#!/bin/bash
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### This script helps setup, run, and cleanup an ebs-scale-test. See README.md
+set -euo pipefail
+BASE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
+## Overridable environment variables. See README.md
+export CLUSTER_TYPE
+export TEST_TYPE
+export REPLICAS
+export DRIVER_VALUES_FILEPATH
+
+export CLUSTER_NAME
+export EXPORT_DIR
+export S3_BUCKET
+export SCALABILITY_TEST_RUN_NAME
+
+CLUSTER_TYPE=${CLUSTER_TYPE:="pre-allocated"}
+TEST_TYPE=${TEST_TYPE:="scale-sts"}
+REPLICAS=${REPLICAS:=1000}
+DRIVER_VALUES_FILEPATH=${DRIVER_VALUES_FILEPATH:="$BASE_DIR/helpers/cluster-setup/scale-driver-values.yaml"}
+# TODO Q: Does anyone need an $OVERRIDE_KUBECONFIG? Let's discuss.
+
+CLUSTER_NAME=${CLUSTER_NAME:="ebs-scale-$CLUSTER_TYPE"}
+EXPORT_DIR=${EXPORT_DIR:=$(mktemp -d)}
+S3_BUCKET=${S3_BUCKET:="ebs-scale-tests"}
```

Consider removing the default here and throwing error if not provided? Considering S3 buckets name as globally unique.

---

I prefer setting a reasonable default instead of putting the burden of setting S3_BUCKET for each run on every user. Though I'm open to making S3_BUCKET a required parameter if other reviewers agree. 

Would choosing a rarer default bucket name satisfy your concern? Or clearer documentation? Something like `ebs-scale-tests-default-bucket-name` is even more unlikely to already be a bucket in user's account.

---

what about `ebs-csi-driver-scale-test-bucket` in case there are other "ebs-scale-tests" buckets out there

---

@rdpsin I just realized your point. Yes lets remove the default and throw an error if not provided. 

Alternatively we can append something like account-id, but I think you are right it's cleanest just to error out. 

@ElijahQuinones there can only be one bucket named `ebs-csi-driver-scale-test-bucket` across all AWS accounts. Unless we generate a unique name by default on behalf of user, they will get err `BucketAlreadyExists` when we attempt to create bucket.

---

I think a pragmatic solution here would be following what we do for the kOps bucket used in CI: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/b28e70ded7863ce6382671b601c3722848ab92f2/hack/e2e/config.sh#L45

Prefixing the bucket name with an account ID practically guarantees that it is unique.

---

Sounds good, either of the options.

---

Addressed in next revision.